src/main/java/com/iexec/sms/secret/web3/Web3SecretRepository.java

```diff
@@ -16,13 +16,12 @@
 
 package com.iexec.sms.secret.web3;
 
-
 import org.springframework.data.repository.CrudRepository;
 
 import java.util.Optional;
 
-public interface Web3SecretRepository extends CrudRepository<Web3Secret, String> {
-
-    Optional<Web3Secret> findWeb3SecretByAddress(String secretAddress);
-
+public interface Web3SecretRepository extends CrudRepository<Web3Secret, Web3SecretHeader> {
+    default Optional<Web3Secret> find(String secretAddress) {
+        return findById(new Web3SecretHeader(secretAddress));
+    }
```

Comment:
```suggestion
    Optional<Web3Secret> findByAddress(String secretAddress);
```
`Collar findBySerialIdAndPet(Long serialId, Pet pet); `
https://www.jpa-buddy.com/blog/the-ultimate-guide-on-composite-ids-in-jpa-entities/

Reply:
Wouldn't it be `findByWeb3SecretHeader_Address`?

Reply:
We can stick to your current implementation and see how it goes :)